src/Functions/FunctionsLanguageClassification.cpp

```diff
@@ -0,0 +1,215 @@
+    {
+        if (code_string.ends_with("-Latn"))
+            code_string.remove_suffix(code_string.size() - 5);
+
+        if (code_string.ends_with("-Hant"))
+            code_string.remove_suffix(code_string.size() - 5);
+
+        // Old deprecated codes
+        if (code_string == "iw")
+            return "he";
+
+        if (code_string == "jw")
+            return "jv";
+
+        if (code_string == "in")
+            return "id";
+
+        if (code_string == "mo")
+            return "ro";
+
+        // Some languages do not have 2 letter codes, for example code for Cebuano is ceb
+        if (code_string.size() != 2)
+            return "other";
+
+        return String(code_string);
+    }
+
+    static void constant(const String & data, String & res)
+    {
+        bool is_reliable = true;
+        const char * str = data.c_str();
+        auto lang = CLD2::DetectLanguage(str, strlen(str), true, &is_reliable);
+        res = codeISO(LanguageCode(lang));
+    }
+
+
+    static void vector(
+        const ColumnString::Chars & data,
+        const ColumnString::Offsets & offsets,
+        ColumnString::Chars & res_data,
+        ColumnString::Offsets & res_offsets)
+    {
+        res_data.reserve(1024);
+        res_offsets.resize(offsets.size());
+
+        size_t prev_offset = 0;
+        size_t res_offset = 0;
+
+        for (size_t i = 0; i < offsets.size(); ++i)
+        {
+            const char * str = reinterpret_cast<const char *>(&data[prev_offset]);
+            String res;
+            bool is_reliable = true;
+
+            auto lang = CLD2::DetectLanguage(str, strlen(str), true, &is_reliable);
+            res = codeISO(LanguageCode(lang));
+
+            size_t cur_offset = offsets[i];
+
+            res_data.resize(res_offset + res.size() + 1);
+            memcpy(&res_data[res_offset], res.data(), res.size());
+            res_offset += res.size();
+
+            res_data[res_offset] = 0;
+            ++res_offset;
+
+            res_offsets[i] = res_offset;
+            prev_offset = cur_offset;
+        }
+    }
+
+
+};
+
+class LanguageClassificationMixedDetect : public IFunction
+{
+public:
+    static constexpr auto name = "detectLanguageMixed";
+
+    static FunctionPtr create(ContextPtr) { return std::make_shared<LanguageClassificationMixedDetect>(); }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 1; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (!isString(arguments[0]))
+            throw Exception(
+                "Illegal type " + arguments[0]->getName() + " of argument of function " + getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
```

Comment:
Which one is legal? Add it to exception message

Reply:
Also you can use fmt
`throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of argument of function {}. Expected String or FixedString", arguments[0]->getName(), getName());`

Reply:
Btw is FixedString supported and why not?